### PR TITLE
Add slogAdapter

### LINKED ISSUE #6402: Add a slogAdapter converting go-kit/log to slog
The Prometheus v3.0.0 uses a `slog` instead of the `go-kit/log`. In other words, the passed log parameter is changed to `slog`.
To upgrade Prometheus v3.0.0, we should pass `slog` instead of `go-kit/log`.

So, it might be good to implement a `slogAdapter` which converts `go-kit/log` to `slog`.

### PR DESCRIPTION
**Pull request by SungJin1212**
This PR adds a slogAdapter converting `go-kit/log` to `slog`.

It would be good to use `slogAdapter` to upgrade the Prometheus version. We could implement features included in Prometheus v3.0.0, like OTLP metadata, OOO ingestion native histogram, UTF-8 support, and so on.

After that, we could migrate whole logs to `slog` if needed.

**Which issue(s) this PR fixes**:
Fixes #6402 

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by yeya24** [comment]
This doesn't seem relevant.

```
==================
WARNING: DATA RACE
Write at 0x0000037316f7 by goroutine 39:
  github.com/cortexproject/cortex/pkg/querier/tripperware.NewQueryTripperware()
      /__w/cortex/cortex/pkg/querier/tripperware/roundtrip.go:112 +0x7a
  github.com/cortexproject/cortex/pkg/querier/tripperware/queryrange.TestRoundTrip()
      /__w/cortex/cortex/pkg/querier/tripperware/queryrange/query_range_middlewares_test.go:71 +0x42b
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1690 +0x226
  testing.(*T).Run.gowrap1()
      /usr/local/go/src/testing/testing.go:1743 +0x44

Previous read at 0x0000037316f7 by goroutine 72:
  github.com/prometheus/prometheus/promql/parser.(*parser).newAggregateExpr()
      /__w/cortex/cortex/vendor/github.com/prometheus/prometheus/promql/parser/parse.go:450 +0x1aa
  github.com/prometheus/prometheus/promql/parser.(*yyParserImpl).Parse()
      /__w/cortex/cortex/vendor/github.com/prometheus/prometheus/promql/parser/generated_parser.y.go:1072 +0x1568
  github.com/prometheus/prometheus/promql/parser.(*parser).parseGenerated()
      /__w/cortex/cortex/vendor/github.com/prometheus/prometheus/promql/parser/parse.go:895 +0xc4
  github.com/prometheus/prometheus/promql/parser.(*parser).ParseExpr()
      /__w/cortex/cortex/vendor/github.com/prometheus/prometheus/promql/parser/parse.go:100 +0xcf
  github.com/prometheus/prometheus/promql/parser.ParseExpr()
      /__w/cortex/cortex/vendor/github.com/prometheus/prometheus/promql/parser/parse.go:171 +0xda
  github.com/cortexproject/cortex/pkg/querier/tripperware/queryrange.Test_evaluateAtModifier.func1()
      /__w/cortex/cortex/pkg/querier/tripperware/queryrange/split_by_interval_test.go:398 +0x1f9
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1690 +0x226
  testing.(*T).Run.gowrap1()
      /usr/local/go/src/testing/testing.go:1743 +0x44

Goroutine 39 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:1743 +0x825
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:2168 +0x85
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1690 +0x226
  testing.runTests()
      /usr/local/go/src/testing/testing.go:2166 +0x8be
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:2034 +0xf17
  main.main()
      _testmain.go:103 +0x164

Goroutine 72 (finished) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:1743 +0x825
  github.com/cortexproject/cortex/pkg/querier/tripperware/queryrange.Test_evaluateAtModifier()
      /__w/cortex/cortex/pkg/querier/tripperware/queryrange/split_by_interval_test.go:388 +0x233
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1690 +0x226
  testing.(*T).Run.gowrap1()
      /usr/local/go/src/testing/testing.go:1743 +0x44
==================
--- FAIL: TestSplitQuery (0.00s)
    --- FAIL: TestSplitQuery/1 (0.01s)
        testing.go:[139](https://github.com/cortexproject/cortex/actions/runs/12193243541/job/34015175708?pr=6405#step:6:140)9: race detected during execution of test
--- FAIL: TestConstSplitter_generateCacheKey (0.00s)
    testing.go:1399: race detected during execution of test
--- FAIL: TestRoundTrip (0.06s)
    testing.go:1399: race detected during execution of test
--- FAIL: TestMergeAPIResponses (0.00s)
    testing.go:1399: race detected during execution of test
--- FAIL: TestResponseWithStats (0.00s)
    testing.go:1399: race detected during execution of test
--- FAIL: TestShouldCache (0.00s)
    --- FAIL: TestShouldCache/negative_offset_on_subqueries (0.00s)
        testing.go:1399: race detected during execution of test
--- FAIL: TestResponse (0.00s)
    testing.go:1399: race detected during execution of test
--- FAIL: TestRequest (0.00s)
    testing.go:1399: race detected during execution of test
--- FAIL: TestSplitByDay (0.03s)
    testing.go:1399: race detected during execution of test
--- FAIL: TestHandleHit (0.02s)
    testing.go:1399: race detected during execution of test
--- FAIL: TestNextIntervalBoundary (0.03s)
    testing.go:1399: race detected during execution of test
--- FAIL: TestLimitsMiddleware_MaxQueryLookback (0.00s)
    testing.go:1399: race detected during execution of test
--- FAIL: TestCompressedResponse (0.00s)
    testing.go:1399: race detected during execution of test
--- FAIL: TestResultsCacheMaxFreshness (0.00s)
    --- FAIL: TestResultsCacheMaxFreshness/0 (0.25s)
        testing.go:1399: race detected during execution of test
FAIL
FAIL	github.com/cortexproject/cortex/pkg/querier/tripperware/queryrange	0.442s
```

**Comment by SungJin1212** [comment]
It seems relevant to https://github.com/cortexproject/cortex/pull/6355.
It seems caused by concurrent access to `EnableExperimentalFunctions`.

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@
 * [FEATURE] Chunk Cache: Support multi level cache and add metrics. #6249
 * [FEATURE] Distributor: Accept multiple HA Tracker pairs in the same request. #6256
 * [FEATURE] Ruler: Add support for per-user external labels #6340
+* [ENHANCEMENT] Log: Add a slogAdapter to upgrade Prometheus v3.0.0. #6405
```
Comment (yeya24) [inline CHANGELOG.md:20]: Umm, this PR only adds a SlogAdapter variable and it is not used anywhere. There is no user facing change I think so let's remove the changelog entry

Comment (SungJin1212) [reply CHANGELOG.md]: Delete it and fix race condition.

File: pkg/util/log/slog_adapter.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import (
+	"log/slog"
+
+	"github.com/go-kit/log"
+	sloggk "github.com/tjhop/slog-gokit"
+)
+
+// GoKitLogToSlog convert go-kit/log to slog
+// usage: logutil.GoKitLogToSlog(gokitLogger)
+func GoKitLogToSlog(logger log.Logger) *slog.Logger {
```
Comment (yeya24) [inline pkg/util/log/slog_adapter.go:12]: I wonder if we should name this function to `PrometheusLoggerToSlot` and have another function for `GoKitLogToSlog` which takes an additional level parameter?

**Review by yeya24 — APPROVED** [review]
Thanks!

**Review by friedrichg — APPROVED** [review]

**Pull request merged**